controllers/storagecluster/generate.go

```diff
@@ -83,3 +90,28 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 func generateStorageQuotaName(storageClassName, quotaName string) string {
 	return fmt.Sprintf("ocs-%s-%s", storageClassName, quotaName)
 }
+
+// generateCephFSProviderParameters function generates extra parameters required for provider storage clusters
+func generateCephFSProviderParameters(initData *ocsv1.StorageCluster) map[string]string {
+	if initData.Spec.AllowRemoteStorageConsumers {
+		deviceSetList := initData.Spec.StorageDeviceSets
+		var deviceSet ocsv1.StorageDeviceSet
+		for _, ds := range deviceSetList {
+			if ds.Name == StorageDeviceName {
+				deviceSet = ds
+			}
+		}
```

Comment:
You need to handle the case where you don't find the specific storage device set and return an error

Reply:
Done